msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -0,0 +1,47 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Xamarin.iOS.Tasks {
+	[TestFixture]
+	public class LocalizationStringTest : TestBase {
+		[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
+		[TestCase ("de-DE", "Unbekanntes Imageformat.")]
+		[TestCase ("en-US", "Unknown image format.")]
+		[TestCase ("es-ES", "formato de imagen desconocido.")]
+		[TestCase ("fr-FR", "format d'image inconnu.")]
+		[TestCase ("it-IT", "Formato immagine sconosciuto.")]
+		[TestCase ("ja-JP", "の読み込みでエラーが発生しました: 画像の形式が不明です。")]
+		[TestCase ("ko-KR", "을(를) 로드하는 동안 오류 발생: 알 수 없는 이미지 형식입니다.")]
+		[TestCase ("pl-PL", "nieznany format obrazu.")]
+		[TestCase ("pt-BR", "formato de imagem desconhecido.")]
+		[TestCase ("ru-RU", "неизвестный формат изображения.")]
+		[TestCase ("tr-TR", "yüklenirken hata oluştu: Görüntü biçimi bilinmiyor.")]
+		[TestCase ("zh-CN", "时出错: 未知图像格式")]
+		[TestCase ("zh-TW", "時發生錯誤: 未知的映像格式。")]
+		public void AllSupportedTranslations (string culture, string errorMessage)
+		{
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+			CultureInfo newCulture;
+			try {
+				newCulture = new CultureInfo (culture);
+				Thread.CurrentThread.CurrentUICulture = newCulture;
+				var task = CreateTask<CollectITunesArtwork> ();
+				task.ITunesArtwork = new TaskItem [] { new TaskItem (Assembly.GetExecutingAssembly ().Location) };
+
+				Assert.IsFalse (task.Execute (), "Execute failure");
+				Assert.AreEqual (1, Engine.Logger.ErrorEvents.Count, "ErrorCount");
+				bool isTranslated = Engine.Logger.ErrorEvents[0].Message.Contains (errorMessage);
+				Assert.IsTrue (isTranslated, culture + ": is not supported correctly. ");
+				Thread.CurrentThread.CurrentUICulture = originalCulture;
+			} catch (CultureNotFoundException) {
+				Thread.CurrentThread.CurrentUICulture = originalCulture;
```

Comment:
You want to put this line in a in a finally cause, not a catch clause:

```csharp
try {
    newCulture = new CultureInfo (culture);
    ...
    Assert.IsTrue (isTranslated, culture + ": is not supported correctly. ");
} finally {
    Thread.CurrentThread.CurrentUICulture = originalCulture;
}
```

also you don't need to catch the CultureNotFoundException (if it's thrown it will cause the test to fail, no need to catch it and then make an assert fail).